lib/seek/json_metadata/data.rb

```diff
@@ -34,6 +34,17 @@ def mass_assign(hash, pre_process: true)
         end
       end
 
+      # extracts all values from the data as a list, including from within nested structures
+      def extract_all_values(data = self)
+        data.values.collect do |value|
+          if value.is_a?(Data)
```

Comment:
Maybe worth using the full namespaced class name here for clarity, since there is a built-in `Data` class in Ruby 3